### PR TITLE
Patch UUID tableprefix

### PR DESCRIPTION
**Pull request by stonecooler**
Fix the following:

A "exposed"-find with contain like this:
`public function edit($uuid = null)
{
  $field = $this->Customers->getExposedKey();
  $customer = $this->Customers->find('exposed', [$field => $uuid])->contain(['Addresses'])->firstOrFail();
  ...
}`

will cause the error "Column 'uuid' in where clause is ambiguous"
Looking at the SQL Log it is clear - no tablename before 'uuid':

`SELECT
  Customers.id AS Customers__id, 
  Customers.address_id AS Customers__address_id, 
  ...
  Customers.uuid AS Customers__uuid, 
  Addresses.id AS Addresses__id, 
  Addresses.organization AS Addresses__organization, 
  ...
  Addresses.uuid AS Addresses__uuid 
FROM 
  customers Customers 
  INNER JOIN addresses Addresses ON Addresses.id = (Customers.address_id) 
WHERE 
  uuid = '29f554bf-f703-4c9a-b777-69925c4064f9'
LIMIT 
  1 `
 
The fix will add the table prefix before 'uuid':
`
SELECT 
  Customers.id AS Customers__id, 
  Customers.address_id AS Customers__address_id, 
  ...
  Customers.uuid AS Customers__uuid, 
  Addresses.id AS Addresses__id, 
  Addresses.organization AS Addresses__organization, 
  ...
  Addresses.nation AS Addresses__nation, 
  Addresses.uuid AS Addresses__uuid 
FROM 
  customers Customers 
  INNER JOIN addresses Addresses ON Addresses.id = (Customers.address_id) 
WHERE 
  Customers.uuid = '29f554bf-f703-4c9a-b777-69925c4064f9' 
LIMIT 
  1
`

### CODE REVIEW

File: src/Model/Behavior/ExposeBehavior.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ public function findExposedList(Query $query, array $options) {
 	 * @return string
 	 */
 	public function getExposedKey(): string {
```
Comment (dereuromark) [inline src/Model/Behavior/ExposeBehavior.php:84]: Should we make this an option?

    $prefixed = false

Then people can use it if needed only.

**Comment by dereuromark** [comment]
The finder key itself must not be prefixed.
See https://github.com/dereuromark/cakephp-expose/pull/3 as alternative.

**Comment by dereuromark** [comment]
How is the plugin working out in general?
And how does it compare to the other existing obfuscate ones/approaches?

**Comment by stonecooler** [comment]
> The finder key itself must not be prefixed.
> See #3 as alternative.

Okay for me. Sorry for deleting my comment, but it was some kind of review - i guess. Too many gui button here I am not yet familiar with.

**Comment by stonecooler** [comment]
> How is the plugin working out in general?
> And how does it compare to the other existing obfuscate ones/approaches?

At the moment im still changing my code from the hashID solution to this new approaches. 
So I will report my experience later. But at the moment I can say:

The nice thing with the old hashID Behavior was: Deactivating the Behavior will not break the Application, because the "find"-Calls are the same. No need to change code. That was very important / helpfull as I moved from 3.8 to 4.
Currently I have a lot to do changing from:

`
$customer = $this->Customers->get($id, [
'contain' => ['Addresses']
]);
`
To:

`$field = $this->Customers->getExposedKey();`
`$customer = $this->Customers->find('exposed', [$field => $uuid])->contain(['Addresses'])->firstOrFail();`

...in many actions

I understand the reason why you changed the obfuscate solution. On the other hand: other solutions are not known to me either.

The first version of your documentation is very good for a beta version of this plugin. Thank you. My first steps with "Migrations" - never used it before. Not everything is clear for me as a partime cake dev, but I can improve it later...

**Comment by stonecooler** [comment]
I use "1.0.0-rc" now and looked at your code changes and don't understand it.

I guessed that this code in my action will break:

```
$field = $this->Customers->getExposedKey();
$customer = $this->Customers->find('exposed', [$field => $uuid])->contain(['Addresses'])->firstOrFail();
```
Because the new flag "$prefixed" is false by default. But the code runs without errors (yes).

With "$field = $this->Customers->getExposedKey(true);" I got the error "The uuid key is required for find('exposed')". Reason: the empty check in 'findExposed' does not use 'getExposedKey', where the table alias is prefixed.

And total not understandable to me: The table alias is always returned 'findExposed':

`return $query->where([$this->getTable()->getAlias() . '.' . $field => $options[$field]]);`

This fixed my main problem in the SQL where the "uuid" has not prefix. Mybe only this change fixed the issue, and we forget "prefix" flag. It seems to me that this flags makes more problems than it solves...or I understand your idea behind the flag not fully.

**Comment by dereuromark** [comment]
Those changes are unrelated.
You shouldnt use any different code, it will now automatically work as you expect it.

**Comment by stonecooler** [comment]
> You shouldnt use any different code, it will now automatically work as you expect it.

Yes. Works great. Thank you.
Reason for the "prefix" flag not understandable to me....but that is okay :-)

**Comment by dereuromark** [comment]
If you want to use the key directly in queries then

$key = ...

$query->where($key => $value)

**Comment by dereuromark** [comment]
I clarified in docs: https://github.com/dereuromark/cakephp-expose/blob/master/docs/README.md#using-the-key-directly-in-queries

**Pull request closed without merging**